### PR TITLE
Added functionality to Support BaselineManagement Module.

### PR DESCRIPTION
**Pull request by bobbytreed**
Added Force Parameter to UserRightsAssignment
Added Functionality to support DynamicAccounts
Added SecuritySetting resource.

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/securitypolicydsc/32)

### CODE REVIEW

File: DSCResources/MSFT_UserRightsAssignment/MSFT_UserRightsAssignment.psm1 [hunk 1]
```diff
@@ -171,19 +182,58 @@ function Set-TargetResource
     }
     else
     {
+        $currRights = Get-TargetResource -Policy $Policy -Identity $Identity
+
+        $Accounts = @()
```
Comment (jcwalker) [inline DSCResources/MSFT_UserRightsAssignment/MSFT_UserRightsAssignment.psm1]: variable names should use camel case

File: DSCResources/MSFT_UserRightsAssignment/MSFT_UserRightsAssignment.psm1 [hunk 2]
```diff
@@ -171,19 +182,58 @@ function Set-TargetResource
...
     {
+        $currRights = Get-TargetResource -Policy $Policy -Identity $Identity
+
+        $Accounts = @()
+        switch ($Identity)
+        {
+            "[Local Account]" { $Accounts += (Get-CimInstance win32_useraccount -Filter "LocalAccount='True'").SID }
+            "[Local Account|Administrator]" 
+            {
+                $AdministratorsGroup = Get-CimInstance -class win32_group -filter "SID='S-1-5-32-544'"
```
Comment (jcwalker) [inline DSCResources/MSFT_UserRightsAssignment/MSFT_UserRightsAssignment.psm1]: variable names should use camel case

File: DSCResources/MSFT_UserRightsAssignment/MSFT_UserRightsAssignment.psm1 [hunk 3]
```diff
@@ -171,19 +182,58 @@ function Set-TargetResource
...
+                $GroupUsers = get-ciminstance -query "select * from win32_groupuser where GroupComponent = `"Win32_Group.Domain='$($env:COMPUTERNAME)'`,Name='$($AdministratorsGroup.name)'`""
```
Comment (jcwalker) [inline DSCResources/MSFT_UserRightsAssignment/MSFT_UserRightsAssignment.psm1]: variable names should use camel case and functions (get-ciminstance) should use pascal case.

File: DSCResources/MSFT_SecurityTemplate/MSFT_SecurityTemplate.psm1 [hunk 4]
```diff
@@ -62,6 +63,7 @@ function Set-TargetResource
     param
     (
         [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
```
Comment (jcwalker) [inline DSCResources/MSFT_SecurityTemplate/MSFT_SecurityTemplate.psm1]: Removed in previous PR

**Comment by GoateePFE** [comment]
How close are we on committing this to master?

**Comment by jcwalker** [comment]
Reviewed 1 of 5 files at r4.
Review status: 1 of 13 files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---

*[DSCResources/MSFT_SecuritySetting/MSFT_SecuritySetting.psm1, line 128 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/32#-Kh2VI2ZrnS16QxWYe3U-r2-128:-Kj30TsoMcROXfDUTjpP:btqlxi8) ([raw file](https://github.com/powershell/securitypolicydsc/blob/3c81260e7b74cb869e6b2c6ee4ec10a198288f60/DSCResources/MSFT_SecuritySetting/MSFT_SecuritySetting.psm1#L128)):*
<details><summary><i>Previously, jcwalker (Jason Walker) wrote…</i></summary><blockquote>

Doesn't follow style guidelines for parameter block format
https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#correct-format-for-parameter-block

</blockquote></details>

Lets put the parameter name on it's own line like the examples in the style guidelines. 

---

*[DSCResources/MSFT_SecuritySetting/MSFT_SecuritySetting.psm1, line 203 at r4](https://reviewable.io:443/reviews/powershell/securitypolicydsc/32#-Kj35Gk4MFoPMbwsKRh5:-Kj35Gk4MFoPMbwsKRh6:bz94i8) ([raw file](https://github.com/powershell/securitypolicydsc/blob/4587268d7b9f9141f22b63db16b288b8c48dd1b4/DSCResources/MSFT_SecuritySetting/MSFT_SecuritySetting.psm1#L203)):*
> ```PowerShell
>     if (@($PSBoundParameters.Keys.Where({$_ -notin "Name", "Ensure"})).Count -eq 0)
>     {
>         Write-Error "No Values Specified!"
> ```

Message not localized

---

*[DSCResources/MSFT_SecuritySetting/MSFT_SecuritySetting.psm1, line 228 at r4](https://reviewable.io:443/reviews/powershell/securitypolicydsc/32#-Kj35CUfjW3fxDOkQysz:-Kj35CUfjW3fxDOkQyt-:blq8rc) ([raw file](https://github.com/powershell/securitypolicydsc/blob/4587268d7b9f9141f22b63db16b288b8c48dd1b4/DSCResources/MSFT_SecuritySetting/MSFT_SecuritySetting.psm1#L228)):*
> ```PowerShell
>             catch
>             {
>                 Write-Error "Unable to set $($keyPair.Key) to $($keyPair.Value)."
> ```

Message is not localized.

---

*[DSCResources/MSFT_SecuritySetting/MSFT_SecuritySetting.psm1, line 263 at r4](https://reviewable.io:443/reviews/powershell/securitypolicydsc/32#-Kj34xPtNtZ7oDP6UsPT:-Kj34xPuCDCxaAW6qk1Z:bcdh875) ([raw file](https://github.com/powershell/securitypolicydsc/blob/4587268d7b9f9141f22b63db16b288b8c48dd1b4/DSCResources/MSFT_SecuritySetting/MSFT_SecuritySetting.psm1#L263)):*
> ```PowerShell
>     "Revision=1" | Out-File $tmpfile -Append
>     
>     $PowerShellProcess = new-object System.Diagnostics.Process
> ```

This block of code is repeated twice.  Can it be put into a function?

---

*[DSCResources/MSFT_SecuritySetting/MSFT_SecuritySetting.psm1, line 268 at r4](https://reviewable.io:443/reviews/powershell/securitypolicydsc/32#-Kj35O2Q3F6RDQrSxflc:-Kj35O2Q3F6RDQrSxfld:bbytit3) ([raw file](https://github.com/powershell/securitypolicydsc/blob/4587268d7b9f9141f22b63db16b288b8c48dd1b4/DSCResources/MSFT_SecuritySetting/MSFT_SecuritySetting.psm1#L268)):*
> ```PowerShell
>     $PowerShellProcess.StartInfo.RedirectStandardOutput = $True
>     $PowerShellProcess.StartInfo.UseShellExecute = $false
>     $PowerShellProcess.start() | Out-Null
> ```

Please change to: $PowerShellProcess.StartInfo.Start()

---

*[DSCResources/MSFT_SecurityTemplate/MSFT_SecurityTemplate.psm1, line 21 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/32#-Kh2VI2Yl1Wrb-Iu8ztT-r2-21:-Kj2nEbYPLSNeu9ckjmY:b-436fs4) ([raw file](https://github.com/powershell/securitypolicydsc/blob/3c81260e7b74cb869e6b2c6ee4ec10a198288f60/DSCResources/MSFT_SecurityTemplate/MSFT_SecurityTemplate.psm1#L21)):*
<details><summary><i>Previously, jcwalker (Jason Walker) wrote…</i></summary><blockquote>

This line was removed in a previous PR.  This is not from the latest branch.
</blockquote></details>

This line appears to be unchanged.

---

*[DSCResources/MSFT_SecurityTemplate/MSFT_SecurityTemplate.psm1, line 66 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/32#-Kh2VI2Yl1Wrb-Iu8ztT-r2-66:-Kj2nMsMyM0hFT-gPdP6:b-436fs4) ([raw file](https://github.com/powershell/securitypolicydsc/blob/3c81260e7b74cb869e6b2c6ee4ec10a198288f60/DSCResources/MSFT_SecurityTemplate/MSFT_SecurityTemplate.psm1#L66)):*
<details><summary><i>Previously, jcwalker (Jason Walker) wrote…</i></summary><blockquote>

Removed in previous PR 
</blockquote></details>

This line appears to be unchanged.

---

*[DSCResources/MSFT_SecurityTemplate/MSFT_SecurityTemplate.psm1, line 116 at r2](https://reviewable.io:443/reviews/powershell/securitypolicydsc/32#-Kh2VI2Yl1Wrb-Iu8ztT-r2-116:-Kj2nZY47xX9XSxfsTnU:b-436fs4) ([raw file](https://github.com/powershell/securitypolicydsc/blob/3c81260e7b74cb869e6b2c6ee4ec10a198288f60/DSCResources/MSFT_SecurityTemplate/MSFT_SecurityTemplate.psm1#L116)):*
<details><summary><i>Previously, jcwalker (Jason Walker) wrote…</i></summary><blockquote>

This line was removed in a previous PR.  This is not from the latest branch.
</blockquote></details>

This line appears to be unchanged.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/securitypolicydsc/32)*

**Pull request merged**